improver/calibration/rainforest_calibration.py

```diff
@@ -397,6 +400,49 @@ def _make_decreasing(self, probability_data: ndarray) -> ndarray:
         )
         return 0.5 * (upper + lower)
 
+    def _evaluate_probabilities(
+        self,
+        forecast_data: ndarray,
+        input_data: ndarray,
+        output_data: ndarray,
+        preprocessing_fun: Optional[Callable[[ndarray], object]] = None,
+    ):
+        """Evaluate probability that error in forecast exceeds thresholds, setting
+        the result to 1 when `forecast + threshold <= 0`.
+
+        Args:
+            forecast_data:
+                1-d containing data for the variable to be calibrated.
+            input_data:
+                2-d array of data for the feature variables of the model
+            output_data:
+                array to populate with output; will be modified in place
+            preprocessing_fun:
+                function to convert data from ndarray to model input format,
+                only used for treelite predictor
+        """
+
+        if preprocessing_fun:
+            input_dataset = preprocessing_fun(input_data)
+        else:
+            input_dataset = input_data
```

Comment:
I've changed the name. I don't think changing the default type to `np.array` would work since `preprocessing_fun` is a Callable.

Reply:
I tested this and it seems to work. The usage here would be to apply `np.array` to an `ndarray` which I believe should return the same value. I'm not sure if there is much/any overhead here but it would remove the need for if/else blocks in this function.

Reply:
Ok, I see what you mean now: the callable is the constructor of np.array. Changed as suggested.

Reply:
I've tested this change and it makes things very slightly slower for lightgbm mode (13:54 minutes vs 13:49 minutes) but I don't think that's a significant problem given that we intend to use treelite for production applications.